RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate Model's folder to Typescript

## Proposed changes

Create interfaces for all files inside the model's folder

## Issue(s)	


## How to test or reproduce


- Reply to a message
- Check if the attachments are normal as default
- Navigate through the app for mobile and tablet devices
- Check if the messages are showing as normal
- When the user is adding another server and presses on the server input, should appear servers and accounts logged before
- Search for messages inside a room
- Open the share view

## Screenshots

## Types of changes




## Checklist



## Further comments
